### PR TITLE
Eliminate use of rewrite, CONST_RATIONAL in ArithMSum

### PR DESCRIPTION
**Pull request by ajreynol**

### CODE REVIEW

File: src/theory/arith/arith_msum.cpp [hunk 1]
```diff
@@ -83,6 +83,7 @@ bool ArithMSum::getMonomialSumLit(Node lit, std::map<Node, Node>& msum)
 {
   if (lit.getKind() == GEQ || lit.getKind() == EQUAL)
   {
+    TypeNode tn = lit[0].getType();
```
Comment (nafur) [inline src/theory/arith/arith_msum.cpp:86]: Please move this further down, probably toward line 100.

File: src/theory/arith/arith_msum.cpp [hunk 2]
```diff
@@ -81,7 +81,8 @@ bool ArithMSum::getMonomialSum(Node n, std::map<Node, Node>& msum)
 
 bool ArithMSum::getMonomialSumLit(Node lit, std::map<Node, Node>& msum)
 {
-  if (lit.getKind() == GEQ || lit.getKind() == EQUAL)
+  if (lit.getKind() == GEQ
+      || (lit.getKind() == EQUAL && lit[0].getType().isRealOrInt()))
```
Comment (nafur) [inline src/theory/arith/arith_msum.cpp:85]: This guards against being called with `lit[0]` being Boolean, right? Shouldn't this rather be checked by the caller?

**Review by nafur — APPROVED** [review]
One comment, otherwise LGTM.

File: src/theory/arith/arith_msum.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/arith/arith_msum.cpp:85]: Well its already guarding for the kind of literal here (GEQ, EQUAL), so this check should go here.

**Pull request merged**